presto-druid/src/main/java/io/prestosql/plugin/druid/DruidJdbcClient.java

```diff
@@ -0,0 +1,365 @@
+                                                                    && Objects.equals(jdbcTableHandle.getTableName(), schemaTableName.getTableName()))
+                                            .collect(Collectors.toList())));
+                }
+                return Optional.of(getOnlyElement(tableHandles));
+            }
+        }
+        catch (SQLException e) {
+            throw new PrestoException(JDBC_ERROR, e);
+        }
+    }
+
+    /**
+     * Overriden since the {@link BaseJdbcClient#getTables(Connection, Optional, Optional)}
+     * method uses character escaping that doesn't work well with Druid's Avatica handler.
+     * Unfortunately, because we can't escape search characters like '_' and '%", this call
+     * ends up retrieving metadata for all tables that match the search
+     * pattern. For ex - LIKE some_table matches somertable, somextable and some_table.
+     * <p>
+     * See {@link DruidJdbcClient#getTableHandle(JdbcIdentity, SchemaTableName)} to look at
+     * how tables are filtered.
+     */
+
+    @Override
+    protected ResultSet getTables(Connection connection, Optional<String> schemaName, Optional<String> tableName)
+            throws SQLException
+    {
+        DatabaseMetaData metadata = connection.getMetaData();
+        return metadata.getTables(DRUID_CATALOG,
+                DRUID_SCHEMA,
+                tableName.orElse(null),
+                null);
+    }
+
+    @Override
+    public Optional<ColumnMapping> toPrestoType(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
+    {
+        switch (typeHandle.getJdbcType()) {
+            case Types.VARCHAR:
+                int columnSize = typeHandle.getColumnSize();
+                if (columnSize > VarcharType.MAX_LENGTH || columnSize == -1) {
+                    return Optional.of(varcharColumnMapping(createUnboundedVarcharType()));
+                }
+                return Optional.of(varcharColumnMapping(createVarcharType(columnSize)));
+            default:
+                return super.toPrestoType(session, connection, typeHandle);
+        }
+    }
+
+    /**
+     * Druid doesn't like table names to be qualified with catalog names in the SQL query.
+     * Hence, overriding this method to pass catalog as an empty string.
+     */
+    @Override
+    public PreparedStatement buildSql(ConnectorSession session, Connection connection, JdbcSplit split, JdbcTableHandle table, List<JdbcColumnHandle> columns)
+            throws SQLException
+    {
+        String schemaName = table.getSchemaName();
+        checkArgument(schemaName.isEmpty() || schemaName.equals("druid"), "Only \"druid\" schema is supported");
+        return new QueryBuilder(identifierQuote)
+                .buildSql(
+                        this,
+                        session,
+                        connection,
+                        "",
```

Comment:
i think null would be more appriopriate

```suggestion
                        null,
```

(please also update method comment)